include/aws/http/request_response.h

```diff
@@ -546,6 +587,20 @@ struct aws_input_stream *aws_http_message_get_body_stream(const struct aws_http_
 AWS_HTTP_API
 void aws_http_message_set_body_stream(struct aws_http_message *message, struct aws_input_stream *body_stream);
 
+/**
+ * Submit a chunk of data to a stream for writing to an outbound HTTP/1.1 connection using chunked transfer encoding.
+ * Note: The stream does not take ownership of allocated resources in the chunk, if any. The caller is responsible
+ * for managing these resources. The data inside the chunk is expected not to be modified by the caller until
+ * the on_complete callback inside the chunk structure is invoked (if provided by the caller). Upon invocation of
+ * on_complete, the caller may release or modify structures in the chunk.
+ * On successful submission of the chunk to the stream, AWS_OP_SUCCESS is returned and AWS_OP_ERR otherwise.
+ */
+AWS_HTTP_API int aws_http1_stream_write_chunk(
+    struct aws_allocator *allocator,
+    struct aws_http_stream *stream,
+    struct aws_input_stream *chunk_data,
+    struct aws_http1_chunk_options *options);
```

Comment:
[style] move the `aws_input_stream` into the options struct. In functions where we use an options struct, we put all the arguments in there, except the 1st argument (object that function is being called on).

Reply:
done.